cairo/tests/ethereum/crypto/test_alt_bn128.py

```diff
@@ -81,6 +81,20 @@ class TestBNP2:
         def test_bnp2_point_at_infinity(self, cairo_run):
             assert cairo_run("bnp2_point_at_infinity") == BNP2.point_at_infinity()
 
+        def test_BNP2_B(self, cairo_run):
+            assert cairo_run("BNP2_B") == BNP2.B
+
+        @given(p=...)
+        def test_bnp2_init(self, cairo_run, p: BNP2):
+            BNP2(p.x, p.y)
+            assert cairo_run("bnp2_init", BNF2(p.x), BNF2(p.y)) == BNP2(p.x, p.y)
+
+        @given(x=..., y=...)
+        def test_bnp2_init_fails(self, cairo_run, x: BNF2, y: BNF2):
+            assume(x != BNF2.zero() or y != BNF2.zero())
```

Comment:
should this assume that BNP2(x,y) fail instead? randomly chosen x and y _could_ be valid no?

Reply:
the chance for randomly finding a point on the curve is small enough to assume that x and y will not be forming a valid point imo.